File: python/src/Behavioral/Observer/DvdRelease.py
#//DvdRelease - a helper class

class DvdRelease:

		def __init__(self,serialNumber, dvdName, dvdReleaseYear, dvdReleaseMonth, dvdReleaseDay) :
			self.serialNumber = serialNumber
			self.dvdName = dvdName
			self.dvdReleaseYear = dvdReleaseYear
			self.dvdReleaseMonth = dvdReleaseMonth
			self.dvdReleaseDay = dvdReleaseDay

		def __str__(self):
			return '"serialNumber":"{0}","dvdName":"{1}","dvdReleaseYear":{2},"dvdReleaseMonth":{3},"dvdReleaseDay":{4}'.format(self.serialNumber, self.dvdName, self.dvdReleaseYear, self.dvdReleaseMonth, self.dvdReleaseDay)


		def updateDvdRelease (serialNumber, dvdName, dvdReleaseYear, dvdReleaseMonth, dvdReleaseDay) :
			self.serialNumber = serialNumber
			self.dvdName = dvdName
			self.dvdReleaseYear = dvdReleaseYear
			self.dvdReleaseMonth = dvdReleaseMonth
			self.dvdReleaseDay = dvdReleaseDay


		def updateDvdReleaseDate (dvdReleaseYear, dvdReleaseMonth, dvdReleaseDay) :
			self.dvdReleaseYear = dvdReleaseYear
			self.dvdReleaseMonth = dvdReleaseMonth
			self.dvdReleaseDay = dvdReleaseDay

File: ruby/src/Behavioral/Observer/DvdRelease.rb
#//DvdRelease - a helper class

class DvdRelease

		attr_accessor :serialNumber, :dvdName, :dvdReleaseYear, :dvdReleaseMonth, :dvdReleaseDay

		def initialize(serialNumber, dvdName, dvdReleaseYear, dvdReleaseMonth, dvdReleaseDay)
 			@serialNumber = serialNumber
			@dvdName = dvdName
			@dvdReleaseYear = dvdReleaseYear
			@dvdReleaseMonth = dvdReleaseMonth
			@dvdReleaseDay = dvdReleaseDay
		end

		def to_s()
			return "\"serialNumber\":\"#{@serialNumber}\",\"dvdName\":\"#{@dvdName}\",\"dvdReleaseYear\":#{@dvdReleaseYear},\"dvdReleaseMonth\":#{@dvdReleaseMonth},\"dvdReleaseDay\":#{@dvdReleaseDay}"
		end


		def updateDvdRelease (serialNumber, dvdName, dvdReleaseYear, dvdReleaseMonth, dvdReleaseDay)
 			@serialNumber = serialNumber
			@dvdName = dvdName
			@dvdReleaseYear = dvdReleaseYear
			@dvdReleaseMonth = dvdReleaseMonth
			@dvdReleaseDay = dvdReleaseDay
		end


		def updateDvdReleaseDate (dvdReleaseYear, dvdReleaseMonth, dvdReleaseDay)
 			@dvdReleaseYear = dvdReleaseYear
			@dvdReleaseMonth = dvdReleaseMonth
			@dvdReleaseDay = dvdReleaseDay
		end

end
